### PR TITLE
datafile wip

### PR DESCRIPTION
**Pull request by jdwyah**
localhost_3010_docs_explanations_concepts_testing (1)

### CODE REVIEW

File: docs/explanations/concepts/testing.md [hunk 1]
```diff
@@ -1,56 +1,65 @@
...
+title: Testing & DataFiles
+sidebar_label: Testing & DataFiles
 sidebar_position: 5
 ---
+:::tip
+Our SDKs and Clients have library and language-specific testing advice. For specific details, refer to the docs for the SDK or client you're using.
+:::
 
-## Testing
+Testing is a first-class citizen in Prefab. We've think we've designed Prefab in a way that makes it easy to test your Prefab using code.
```
Comment (jkebinger) [inline docs/explanations/concepts/testing.md:10]: Prefab-using should be hyphenated i think?

File: docs/explanations/concepts/testing.md [hunk 2]
```diff
@@ -84,3 +93,44 @@ Don't use the `PrefabProvider`. Instead, use the `PrefabTestProvider` and pass i
...
+:::
+
+### Keeping The Datafile Up To Date
+The CLI download will take a snapshot of the configuration at a given moment, but it won't keep it up to date. As you add configuration or feature flags that you need to test, you'll need to update the datafile by re-running the CLI command and committing the new datafile.
+
+### Best Practices For Testing
+
+The best practice for testing is to create a test environment and use a datafile for the bulk of your configuration. 
+
+You can then use mocking to override specific values as needed, when you are testing the behavior of a specific feature flag or config.
```
Comment (semanticart) [inline docs/explanations/concepts/testing.md:136]: Maybe you want to lead with this?

Comment (jdwyah) [reply docs/explanations/concepts/testing.md]: great point

File: docs/explanations/concepts/testing.md (hunk 1 above)
Comment (jdwyah) [reply docs/explanations/concepts/testing.md]: 👍 tweaked wording to avoid this.

**Pull request merged**